assembly/src/assembler/mod.rs

```diff
@@ -796,7 +796,7 @@ impl Assembler {
                         let split_node =
                             MastNode::new_split(then_blk, else_blk, mast_forest_builder.forest());
 
-                        mast_forest_builder.ensure_node(split_node)
+                        mast_forest_builder.ensure_node(split_node).map_err(AssemblyError::from)?
```

Comment:
Is `.map_err()` here (and in other similar places) necessary? I would have thought that the compiler would be able to infer the conversion automatically and just `mast_forest_builder.ensure_node(split_node)?` would have been enough.

Reply:
Updated. Just noting that the conversion required MastForestError to implement the miette::Diagnostic trait to for this. So the core crate now imports miette.

Reply:
Ah! In that case, I'd probably prefer to keep the extra `map_err()` here and avoid an additional import.